tests/src/Interop/PInvoke/Attributes/LCID/LCIDNative.cpp

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <stdio.h>
+#include <windows.h>
+#include <tchar.h>
+#include <xplatform.h>
+
+char* strManaged = "Managed\0String\0";
+int   lenstrManaged = 7; // the length of strManaged
+
+char* strReturn = "a\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
+char* strFalseReturn = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
+
+char* strNative = " Native\0String\0";
+int lenstrNative = 7; //the len of strNative
+
+//Test Method1
+
+extern "C" LPSTR ReturnString()
+{
+    size_t lenstrReturn = strlen(strReturn);
+    LPSTR ret = (LPSTR)(CoTaskMemAlloc(sizeof(char)*(lenstrReturn+1)));
+    memset(ret,'\0',lenstrReturn+1);
+    strncpy_s(ret,lenstrReturn + 1,strReturn,1);
+    return ret;
+}
+
+extern "C" LPSTR ReturnFalseString()
+{
+    size_t lenstrFalseReturn = strlen(strFalseReturn);
+    LPSTR ret = (LPSTR)(CoTaskMemAlloc(sizeof(char)*(lenstrFalseReturn+1)));
+    memset(ret,'\0',lenstrFalseReturn+1);
+    strncpy_s(ret,lenstrFalseReturn + 1,strFalseReturn,1);
+    return ret;
+}
+
+//Test Method2
+extern "C" DLL_EXPORT LPSTR Marshal_InOut1(int lcid, LPSTR s)
+{
+    printf("LCID:%d\n\n",lcid);
+
+    //Check the Input
+    size_t len = strlen(s);
+    if((len != lenstrManaged)||(memcmp(s,strManaged,len)!=0))
+    {
+        printf("Error in Function Marshal_InOut(Native Client)\n");
+
+        //Expected
```

Comment:
These expected/actual part are similar and duplicate for all of these functions, could you create a common function and each scenario can call that common method instead?

Reply:
Fixed